test/parallel/test-inspector-open.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+const common = require('../common');
+
+// Test inspector open()/close()/url() API. It uses ephemeral ports so can be
+// run safely in parallel.
+
+const assert = require('assert');
+const fork = require('child_process').fork;
+const net = require('net');
+const url = require('url');
+
+common.skipIfInspectorDisabled();
+
+if (process.env.BE_CHILD)
+  return beChild();
+
+const child = fork(__filename, {env: {BE_CHILD: 1}});
+
+child.once('message', common.mustCall((msg) => {
+  assert.strictEqual(msg.cmd, 'started');
+
+  child.send({cmd: 'open', args: [0]});
+  child.once('message', common.mustCall(firstOpen));
+}));
+
+let firstPort;
+
+function firstOpen(msg) {
+  assert.strictEqual(msg.cmd, 'url');
+  const port = url.parse(msg.url).port;
+  ping(port, (err) => {
+    assert.ifError(err);
+    child.send({cmd: 'open', args: []});
```

Comment:
since in `agent->StartIoThread` first line is `  if (io_ != nullptr)`
It doesn't matter what `args` you pass right?

IMHO put a comment here and document that calling `inspector.open([port[, host[, wait]]])` while inspector agent is running has no effect.